test/collection.js

```diff
@@ -656,6 +656,20 @@ describe('Chained Collection', () => {
     ).to.reject()
   })
 
+  it('reverse', async () => {
+    expect(
+      await Collect([1, 2, 3]).reverse().all()
+    ).to.equal([3, 2, 1])
+
+    expect(
+      await Collect([1]).reverse().all()
+    ).to.equal([1])
+
+    expect(
+      await Collect([]).reverse().all()
+    ).to.equal([])
+  })
```

Comment:
Can you please add a test that the original array isn't changed when applying `.reverse()`?

```suggestion
  })
  
  const items = [1, 2, 3]
  const collection = Collect(items)
  
  expect(
      await collection.reverse().all()
    ).to.equal([3, 2, 1])
  expect(
      await collection.all()
    ).to.equal([1, 2, 3])
```